istioctl/pkg/multicluster/remote_secret.go

```diff
@@ -222,23 +227,92 @@ func createRemoteSecretFromTokenAndServer(tokenSecret *v1.Secret, clusterName, s
 	return createRemoteServiceAccountSecret(kubeconfig, clusterName, secName)
 }
 
-func getServiceAccountSecret(kube kubernetes.Interface, saName, saNamespace string) (*v1.Secret, error) {
-	serviceAccount, err := kube.CoreV1().ServiceAccounts(saNamespace).Get(context.TODO(), saName, metav1.GetOptions{})
+func getServiceAccountSecret(kube kubernetes.Interface, opt RemoteSecretOptions) (*v1.Secret, error) {
+	// Create the service account if it doesn't exist.
+	serviceAccount, err := getOrCreateServiceAccount(kube, opt)
 	if err != nil {
 		return nil, err
 	}
+
 	if len(serviceAccount.Secrets) != 1 {
 		return nil, fmt.Errorf("wrong number of secrets (%v) in serviceaccount %s/%s",
-			len(serviceAccount.Secrets), saNamespace, saName)
+			len(serviceAccount.Secrets), opt.Namespace, opt.ServiceAccountName)
 	}
+
 	secretName := serviceAccount.Secrets[0].Name
 	secretNamespace := serviceAccount.Secrets[0].Namespace
 	if secretNamespace == "" {
-		secretNamespace = saNamespace
+		secretNamespace = opt.Namespace
 	}
 	return kube.CoreV1().Secrets(secretNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
 }
 
+func getOrCreateServiceAccount(kube kubernetes.Interface, opt RemoteSecretOptions) (*v1.ServiceAccount, error) {
+	allSAs, err := kube.CoreV1().ServiceAccounts(opt.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, sa := range allSAs.Items {
+		if sa.Name == opt.ServiceAccountName {
+			// Found it.
+			return &sa, nil
+		}
+	}
+
+	// Create a basic IOP File to install the minimum configuration for the service account.
+	iopFile, err := createServiceAccountIOPFile()
```

Comment:
Because there are several things that have to be configured ... the sa, the clusterrole and the clusterrolebinding. And the set of permissions needs to be in sync with our installation config.

Reply:
I'm uneasy about doing this as it doesn't feel like the right way to go. You're basically taking a whole car and just turning on the radio for the sake of some music :-) Unfortunately I don't have a particularly clean and easy alternative to offer. The best thing I can think of is to extend the helm pkg to select just the templates you need and render the files that way. It's better not to involve all the machinery of GenerateConfig for this.

Reply:
The issue is that this is a 1.8 blocker so we need a solution. What exactly is involved with extending the helm packages as you suggest?